### ISSUE TITLE
The wrong type of TaxCategory SubRate Amount

### ISSUE THREAD

**Issue by xferra** [commercetools/commercetools-dotnet-core-sdk-v2#141]
**Describe the bug**
I suppose there is a typo in data type.

V1 SDK is `decimal`
https://github.com/commercetools/commercetools-dotnet-core-sdk/blob/95538255acd49e16a37f5d809dce03b865b80b2b/commercetools.Sdk/commercetools.Sdk.Domain/TaxCategories/SubRate.cs#L6

V2 SDK is `double`
https://github.com/commercetools/commercetools-dotnet-core-sdk-v2/blob/ecc50189cc277ddd07f2c04dbded8c4a4ac39835/commercetools.Sdk/commercetools.Sdk.Api/Generated/Models/TaxCategories/ISubRate.cs#L11

**To Reproduce**
-

**Expected behavior**
V2 SDK TaxCategory SubRate Amount should be `decimal`.

**Screenshots/Code snippet**

**Stack information (please complete the following information):**
 - dotnet: Core 3.0
 - SDK: v2.2

**Additional context**
When calculating TaxRate using external provider the followng error occurs:

```json
{
	"statusCode": 400,
	"message": "A total amount of 0.09999999999999999 was given, but the total of all subrates is 0.1.",
	"errors": [
		{
			"code": "InvalidOperation",
			"message": "A total amount of 0.09999999999999999 was given, but the total of all subrates is 0.1.",
			"action": {
				"action": "setLineItemTaxRate",
				"lineItemId": "2099ecec-628f-42b9-bcea-8d5d82f3bd5e",
				"externalTaxRate": {
					"name": "Avalara calculated tax",
					"amount": 0.09999999999999999,
					"country": "US",
					"subRates": [
						{
							"name": "IL STATE TAX",
							"amount": 0.0625
						},
						{
							"name": "IL COUNTY TAX",
							"amount": 0.0175
						},
						{
							"name": "IL CITY TAX",
							"amount": 0.01
						},
						{
							"name": "IL SPECIAL TAX",
							"amount": 0.01
						}
					]
				}
			},
			"actionIndex": 1
		}
	]
}
```

**Comment by jenschude** [comment]
Hi,

sorry we missed this issue. Will open a PR in the upstream repository to fix it

**Comment by jenschude** [comment]
A fix has been applied also to all other fields of type double.

We released this as a patch version of the latest major release. Please see:

https://github.com/commercetools/commercetools-dotnet-core-sdk-v2/releases/tag/3.0.1

**Comment by jenschude** [comment]
Just released 3.0.2 with some additional fixes of number types
